### PR TITLE
Write to stderr in ParseAPI::parsing_printf

### PR DESCRIPTION
**Pull request by hainest**
Using a per-thread file here is cumbersome. Including the thread id in the message and synchronizing vprintf makes this work for multiple threads.

### CODE REVIEW

File: parseAPI/src/debug_parse.C [hunk 1]
```diff
@@ -59,23 +57,36 @@ int Dyninst::ParseAPI::parsing_printf_int(const char *format, ...)
...
     va_start(va,format);
-    int ret = vfprintf(log_file, format, va);
-    fflush(log_file);
-    va_end(va);
 
+    auto ret = [&]() {
+      int v{};
+      #pragma omp critical
+      {
+#ifdef __clang__
```
Comment (kupsch) [inline parseAPI/src/debug_parse.C:77]: To preserve the -Wformat-nonliteral, which is a useful security property, I'd suggest removing the creation of a new format string, by just using `fprintf(stderr, "[thread %u] ", id);` and then the `vfprintf(stderr, format, va);` without the `#pragma`'s.

File: parseAPI/src/debug_parse.C [hunk 2]
```diff
@@ -59,23 +57,36 @@ int Dyninst::ParseAPI::parsing_printf_int(const char *format, ...)
...
+        #pragma clang diagnostic push
+        #pragma clang diagnostic ignored "-Wformat-nonliteral"
+#endif
+        v = vfprintf(stderr, msg.c_str(), va);
+#ifdef __clang__
+        #pragma clang diagnostic pop
+#endif
+        fflush(stdout);
```
Comment (kupsch) [inline parseAPI/src/debug_parse.C:85]: `fflush(stderr)` instead of `stdout`.

File: parseAPI/src/debug_parse.C (hunk 1 above)
Comment (hainest) [reply parseAPI/src/debug_parse.C]: Not sure why I didn't think of that... Changed.

File: parseAPI/src/debug_parse.C (hunk 2 above)
Comment (hainest) [reply parseAPI/src/debug_parse.C]: Derp. Fixed.

File: parseAPI/src/debug_parse.C [hunk 3]
```diff
@@ -59,23 +57,28 @@ int Dyninst::ParseAPI::parsing_printf_int(const char *format, ...)
...
-    int ret = vfprintf(log_file, format, va);
-    fflush(log_file);
-    va_end(va);
 
+    auto ret = [&]() {
+      int v{};
+      #pragma omp critical
+      {
+        v = fprintf(stderr, "[thread %d] ", id);
+        v &= vfprintf(stderr, format, va);
```
Comment (kupsch) [inline parseAPI/src/debug_parse.C:76]: The `&=` doesn't make sense.  fprintf returns the number of characters written or -1 on error.  You should record each return value separately in `r1` and `r2` with a return value of `r1 != -1 && r2 != -1) ? (r1 + r2) : -1`.  Slightly better would be to also only execute the second `vprintf` if `r1 != -1`.  It is also likely that nothing looks at the return value.

**Review by kupsch — CHANGES_REQUESTED** [review]
slight change in error handling.

**Review by kupsch — APPROVED** [review]

**Pull request merged**